datastruc_ex10.py

```diff
@@ -26,11 +19,62 @@ def cyclicShift(str1: str, str2: str):
     else:
         return False
 
+# option 2
+
+
+def rotate(str: str, d: int):
+    '''
+    A function which returns a new string shifted by d length
+    '''
+    first = str[0: len(str)-d]
+    second = str[len(str)-d:]
+    return second + first
+
+
+def cyclicShift2(str1: str, str2: str):
+    '''
+    A function which recieves 2 strings and returns
+    True is str2 is a "cyclic shift" of str1 and False otherwise
+    '''
+    new_str = []
+    for i in range(len(str1)):
+        new_str.append(rotate(str1, i))
+    if str2 in new_str:
+        return True
+    else:
+        return False
+
```

Comment:
This implementation in inefficient. I wont accept it. 
1. I dont to check all the possible shifts before returning an answer. I want you to return true upon finding a solution.
2.  There is no need to keep al created rotations in a list.  create a new string -> check it -> decide what to do

Reply:
done